src/core/lib/iomgr/tcp_windows.cc

```diff
@@ -252,13 +259,15 @@ static void win_read(grpc_endpoint* ep, grpc_slice_buffer* read_slices,
     gpr_log(GPR_INFO, "TCP:%p win_read", tcp);
   }
 
-  if (tcp->shutting_down) {
+  gpr_mu_lock(&tcp->read_mu);
```

Comment:
I don't believe this lock is necessary here. An endpoint should never have more than one thread concurrently calling `grpc_tcp_read`, and because of that, there's no chance that any of this read state could be accessed concurrently. Race potential comes after the on_read callback is registered, which is the last line of this function.